kivy/ext/mpl/backend_kivy.py

```diff
@@ -0,0 +1,1013 @@
+            if len(rgbFace) == 4 and rgbFace[3] != 1.0 and not forced_alpha:
+                attrib['fill-opacity'] = str(rgbFace[3])
+
+        if forced_alpha and self.get_alpha() != 1.0:
+            attrib['opacity'] = str(self.get_alpha())
+
+        offset, seq = self.get_dashes()
+        if seq is not None:
+            attrib['line-dasharray'] = ','.join(['%f' % val for val in seq])
+            attrib['line-dashoffset'] = six.text_type(float(offset))
+
+        linewidth = self.get_linewidth()
+        if linewidth:
+            rgb = self.get_rgb()
+            attrib['line'] = str(rgb)
+            if not forced_alpha and rgb[3] != 1.0:
+                attrib['line-opacity'] = str(rgb[3])
+            if linewidth != 1.0:
+                attrib['line-width'] = str(linewidth)
+            if self.get_joinstyle() != 'round':
+                attrib['line-linejoin'] = self.get_joinstyle()
+            if self.get_capstyle() != 'butt':
+                attrib['line-linecap'] = _capstyle_d[self.get_capstyle()]
+
+        return attrib
+
+
+def draw_if_interactive():
+    '''Handle whether or not the backend is in interactive mode or not.
+    '''
+    if matplotlib.is_interactive():
+        figManager = Gcf.get_active()
+        if figManager:
+            figManager.canvas.draw_idle()
+
+
+class Show(ShowBase):
+    '''mainloop needs to be overwritten to define the show() behavior for kivy
+       framework.
+    '''
+    def mainloop(self):
+        app = App.get_running_app()
+        if app:
+            app.run()
+
+show = Show()
+
+
+def new_figure_manager(num, *args, **kwargs):
+    '''Create a new figure manager instance for the figure given.
+    '''
+    # if a main-level app must be created, this (and
+    # new_figure_manager_given_figure) is the usual place to
+    # do it -- see backend_wx, backend_wxagg and backend_tkagg for
+    # examples. Not all GUIs require explicit instantiation of a
+    # main-level app (egg backend_gtk, backend_gtkagg) for pylab
+    FigureClass = kwargs.pop('FigureClass', Figure)
+    thisFig = FigureClass(*args, **kwargs)
+    return new_figure_manager_given_figure(num, thisFig)
+
+
+def new_figure_manager_given_figure(num, figure):
+    '''Create a new figure manager instance for the given figure.
+    '''
+    canvas = FigureCanvasKivy(figure)
+    canvas.draw()
+    manager = FigureManagerKivy(canvas, num)
+    _create_App(canvas, manager.toolbar.actionbar)
+    return manager
+
+
+class FigureCanvasKivy(FocusBehavior, Widget, FigureCanvasBase):
+    '''FigureCanvasKivy class. See module documentation for more information.
+
+    .. versionadded:: 1.9.1
+    '''
+
+    def __init__(self, figure, **kwargs):
+        Window.bind(mouse_pos=self._on_mouse_pos)
+        self.bind(size=self._on_size_changed)
+        self.inside_figure = True
+        self.figure = figure
+        super(FigureCanvasKivy, self).__init__(figure=self.figure, **kwargs)
+        self.focus = True
```

Comment:
Yes, that's the point. Imagine you start off with a textinput as well as a mpl widget. Say you want the textinput to have focus, but then the mpl widget will steal the focus. The better way is to not set focus and instead by the user clicking on the widget it gets focus - that's how focus works in kivy.